napari/components/_tests/test_viewer_model.py

```diff
@@ -358,7 +358,7 @@ def test_swappable_dims():
     image_data = np.random.random((7, 12, 10, 15))
     image_name = viewer.add_image(image_data).name
     assert np.all(
-        viewer.layers[image_name]._data_view == image_data[3, 6, :, :]
+        viewer.layers[image_name]._data_view == image_data[4, 6, :, :]
```

Comment:
Given that 7 is odd and we want to slice at the midpoint, shouldn't this always be 3 instead of 4 regardless of rounding? I know this test is testing something other than the mid-point initialization, but still feels wrong.

Reply:
Oh, I think actually this explains why we were having [this issue with an extra slice](https://github.com/napari/napari/pull/5582#issuecomment-1507050701).

When we merged #5522, the interaction between `point` and `range` changed slightly because the source of truth used to be `current_step`, which is integers. This, combined with the idea that `range` should reflect the builtin `range()` and thus have the `stop` be exclusive, meant that `dims.point` was effectively rounding down in these cases.

Now that `point` is a float, `current_step` is computed from it, and we should probably round down sowhere where we didnt?

I'm not entirely sure this is the issue, but it makes sense atm :P